### ISSUE TITLE
[Bug]AR Foundation 4.0.8 rejected from App Store (TrueDepth API's)

### ISSUE THREAD

**Issue by sergiosolorzano** [Unity-Technologies/arfoundation-samples#623]
Hello

Duplicate of this closed issue:

[#477](https://github.com/Unity-Technologies/arfoundation-samples/issues/477) and re-opened [#528](https://github.com/Unity-Technologies/arfoundation-samples/issues/528#)

Similar to the other cases App using ARFoundation was rejected by apple today.  The app doesn't use TrueDepth.

"During review, we found that your app includes TrueDepth APIs. However, we were not able to locate any features in your app that use TrueDepth APIs. If your app uses the Unity ARKit plugin, it would be appropriate to [update to the latest version](https://forum.unity.com/threads/submitting-arkit-apps-to-appstore-without-face-tracking.504572/#post-3297235), which includes a setting that allows you to exclude TrueDepth APIs."

Findings:
1) I don't see [ARKitSettings ](https://forum.unity.com/threads/submitting-arkit-apps-to-appstore-without-face-tracking.504572/#post-3297235)equivalent in ARFoundation to exclude TrueDepth, does it exist?

2) Following @mdurand42 's and [recommendation ](https://github.com/Unity-Technologies/arfoundation-samples/issues/477) and 
@tdmowrer [here](https://github.com/Unity-Technologies/arfoundation-samples/issues/528#) I've also checked the Xcode project generated by Unity and indeed it does include a Facetracking library libUnityARKitFaceTracking.a

![image](https://user-images.githubusercontent.com/24430655/95940386-16b42800-0dd6-11eb-8337-fd13d9b34187.png)

3) Package includes ARKit Face Tracking
![image](https://user-images.githubusercontent.com/24430655/95940470-50852e80-0dd6-11eb-8441-20bded09a96d.png)

Should I remove the ARKit Face tracking package or maybe that [breaks ARF](https://forum.unity.com/threads/app-rejected-for-using-truedepth-api.780242/)? Or should I just a raw right-click & delete the ARKitFaceTracking library libUnityARKitFaceTracking.a from XCode project and leave the ARKit Face tracking package as is?

Please give me some guidance what a good next step to resolve this is please.

Thank you

**Comment by DavidMohrhardt** [comment]
You can remove the package and still use ARFoundation.  You just will not be able to access the ARKit face tracking provider if the package is removed (which relies on a native plugin to interact with the ARKit true Depth API).  As long as you do not need face tracking in your application you can do this without any problem.

**Comment by sergiosolorzano** [comment]
Hi @DavidMohrhardt , thanks a bunch for your prompt response!

After experimenting and as you kindly highlight, removing ARKit face tracking package didn't break the project , though I note I did have to delete script ARKitBlendShapeVisualizer.cs as mentioned by [@posthaugness and further confirmed by @SinSeia in unity's forum](https://forum.unity.com/threads/app-rejected-for-using-truedepth-api.780242/) to be able to build.

I was then able to upload to apple now awaiting feedback. I'll post back when I hear :)

thank you again for your help !

**Comment by tdmowrer** [comment]
@sergiosolorzano This is exactly the reason we provide face tracking functionality via a separate package. This is not a bug, but the expected behavior. Please see https://github.com/Unity-Technologies/arfoundation-samples#why-is-arkit-face-tracking-a-separate-package for an explanation.

As @DavidMohrhardt says, if you are not using face tracking, then you should remove the ARKit Face Tracking package. Its presence triggers additional privacy concerns during app validation. See [Apple's statement](https://developer.apple.com/documentation/arkit/arfacetrackingconfiguration?language=objc):

> **Note**
>
> Face tracking provides your app with personal facial information. If you use ARKit face tracking features, your app must include a privacy policy describing to users how you intend to use face tracking and face data. For details, see the [Apple Developer Program License Agreement](https://developer.apple.com/terms/).

**Comment by sergiosolorzano** [comment]
Thank you for the clarification @tdmowrer , now I undertand and makes sense! 
Cheers !

**Comment by ACWesterberg** [comment]
Evening people! So I had this issue with Apple to but they sent me a link on how to resolve this. You can select or deselect using Face Tracking in ARKit Settings that can be found in the XR Plug-in Management. Just uncheck face tracking in the inspector. More info see here https://forum.unity.com/threads/submitting-arkit-apps-to-appstore-without-face-tracking.504572/#post-3297235

**Comment by tdmowrer** [comment]
That forum link is not talking about the ARKit XR Plugin that is part of ARFoundation, but yes, you can disable face tracking from XR Plug-in Management. This is exactly the reason it is there :)

**Comment by ACWesterberg** [comment]
> That forum link is not talking about the ARKit XR Plugin that is part of ARFoundation, but yes, you can disable face tracking from XR Plug-in Management. This is exactly the reason it is there :)

my apologies. I had the issue with TrueDepth API from my submission today and got rejected for it. Hopefully ticking that box solves the issue

**Comment by tdmowrer** [comment]
The way to be sure is to check your Xcode project and make sure it does _not_ reference `libUnityARKitFaceTracking.a`, as in this screenshot:

![Screen Shot 2020-10-28 at 5 11 09 PM](https://user-images.githubusercontent.com/4934281/97509906-bcbc7200-1940-11eb-90b4-32cec59c14da.png)

If you still see it there, you can remove the ARKit Face Tracking package via the Package Manager.

**Comment by ACWesterberg** [comment]
> The way to be sure is to check your Xcode project and make sure it does _not_ reference `libUnityARKitFaceTracking.a`, as in this screenshot:
> 
> ![Screen Shot 2020-10-28 at 5 11 09 PM](https://user-images.githubusercontent.com/4934281/97509906-bcbc7200-1940-11eb-90b4-32cec59c14da.png)
> 
> If you still see it there, you can remove the ARKit Face Tracking package via the Package Manager.

Yep I did that as per the link, and it's not there! Thank you :)

**Comment by shacharoz** [comment]
excuse me guys for reopening this ticket, but what if I actually DO use the face tracking?
if I would delete the blend shapes file, would that help me?
do I must use the true depth API? what is it anyway?? :)

thanks all!

**Comment by tdmowrer** [comment]
From https://developer.apple.com/documentation/arkit/arfacetrackingconfiguration

> Because face tracking provides your app with personal facial information, your app must include a privacy policy describing to users how you intend to use face tracking and face data. For details, see the Apple Developer Program License Agreement.

**Comment by shacharoz** [comment]
I must say it's a little unclear.
I already have privacy policy in the app.
I would expect that apple would provide some sort of clearer guidelines
here.

Does anybody here already approved a face tracking app with apple and could
help with this point?
I am only using the basic face tracking, for a simple try-on, and I'm not
saving any data. So this is the most basic use of ar foundation or arkit.

**Comment by tdmowrer** [comment]
> I already have privacy policy in the app.

Have you submitted your app to the app store? Apple should tell you if your existing policy is sufficient or not as part of the approval process.

In any case, this probably isn't the right forum for questions on Apple's policies and app store submissions. https://developer.apple.com/support/ might be a better starting point.

**Comment by shacharoz** [comment]
i thought we are doing this together :) :)

ok I get that this is not a bug of ar foundation, but it is related to it... because I am trying to publish an app with basic face tracking, and somehow its been rejected, for a very basic feature. 
i was just wondering if anyone already published a face app and can assist with this. if not, then i can go and try to figure out with apple. but of course this is done in parallel channel...

i do appreciate the assistance anyway.

**Comment by tdmowrer** [comment]
> i thought we are doing this together :) :)

Well, I like to think so too :) but I don't have any advice for you other than what we can both read on Apple's website.

> I am trying to publish an app with basic face tracking, and somehow its been rejected...

Did they give a reason for the rejection?

**Comment by shacharoz** [comment]
let me wait now for their next feedback. i have sent a new build for a review.
i have stripped many features out and I wanna see their response.

they just keep saying that I'm using TrueDepth and I need to remove it. i am now adding note on the privacy document as well.

**Comment by shacharoz** [comment]
here is what they replied this time. i have explained numerous times that I'm not using any interfaces and not saving data. i still did not add the info on the privacy policy (I think it will remove only point A).
the app includes a simple face tracking mechanism as a try-on simulator. the ap already exists on android PlayStore: https://play.google.com/store/apps/details?id=com.hairkonfidence.hair


"
Guideline 2.1 - Information Needed
A. We have started the review of your app, but we are not able to continue because we need additional information about how your app uses face data.

Next Steps:
To help us proceed with the review of your app, please provide complete and detailed responses to the following questions.
- What face data is your app collecting?
- For what purposes are you collecting this information? Please provide a complete and clear explanation of all planned uses of this data.
- Will the data be shared with any third parties? Where will this information be stored?
- Which are the relevant sections of your privacy policy that explain collection, use, disclosure, sharing, and retention of face data?
- Please quote the specific text from your privacy policy concerning face data.


B. We have started the review of your app, but we are not able to continue because we need additional information about how your app uses information collected by the TrueDepth API.

Next Steps:
To help us proceed with the review of your app, please provide complete and detailed information to the following questions.
- What information is your app collecting using the TrueDepth API?
- For what purposes are you collecting this information? Please provide a complete and clear explanation of all planned uses of this data.
- Will the data be shared with any third parties? Where will this information be stored?
"

**Comment by tdmowrer** [comment]
Apple's explanation of what they want and how you are failing to comply with it seem pretty straightforward to me.

> i still did not add the info on the privacy policy

Why not? Isn't this what they are asking for?

> the ap already exists on android PlayStore

I doubt that Apple will consider this relevant :)

Also (since you asked in your original question), the "TrueDepth" camera refers to the front-facing depth camera that is used by ARKit when performing face tracking. Generally, "TrueDepth API" means "any API that involves face tracking".

**Comment by shacharoz** [comment]
added here the android app, just for you guys as a reference to the try on features we created.

> update of privacy policy

done by another person, and we were delayed because of new year's. working on it.

> the "TrueDepth" camera refers to the front-facing depth camera that is used by ARKit when performing face tracking. Generally, "TrueDepth API" means "any API that involves face tracking".

ahh, thats new info for me. so does that mean that i am using the TruDepth, even if i "dont want to"? do i even have a choice? how can i explain to apple that i am not using it directly or through AR Foundation?
do you think it would just be enough to add a specific note on the privacy policy about this as well?

what do you think about this text:
" the application uses face tracking with Apple TrueDepth capabilities for the basic augmented reality try-on.
the application does not collect any data on the user face. 
the application does not save any information on the user or the user face or any other data from that aspect.
the application does not use TrueDepth API to collect information.
"

**Comment by tdmowrer** [comment]
> does that mean that i am using the TrueDepth, even if i "dont want to"? do i even have a choice?

"TrueDepth" is just a trademark for their face tracking technology. So yes, you have a choice: if you don't want to use it, then don't use face tracking. But if you want face tracking via ARKit, then you have to use TrueDepth.

> how can i explain to apple that i am using it through AR Foundation, but not directly?

What matters is whether your app uses face tracking.

> what do you think about this text...

This is not the right forum for advice on the specific language required by Apple's privacy policy. I encourage you to ask them.

**Comment by shacharoz** [comment]
ok. thanks!
at least now i am aware that i am using TrueDepth...
i have sent this text to them, and i will update with their answer.

thansk Tim
